### PR TITLE
Code cleanup: protect all copy constructors

### PR DESCRIPTION
**Pull request by Susucre**
So this is a big PR.

The first commit is only applying the following replacement regex on all codebase:
regex

The second commit is removing the redundant copy that were made using those. And add a few copy methods to class missing them.

### CODE REVIEW

File: Mage/src/main/java/mage/game/match/MatchPlayer.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public MatchPlayer(Player player, Deck deck, Match match) {
      * @param newPlayer
      * @return
      */
-    protected MatchPlayer(final MatchPlayer source, Player newPlayer) {
+    public MatchPlayer(final MatchPlayer source, Player newPlayer) {
```
Comment (Susucre) [inline Mage/src/main/java/mage/game/match/MatchPlayer.java]: the only false positive of the regex.

File: Mage.Sets/src/mage/cards/s/SurgeNode.java [hunk 2]
```diff
@@ -18,23 +17,24 @@
...
+        this.addAbility(new EntersBattlefieldAbility(new AddCountersSourceEffect(CounterType.CHARGE.createInstance(6)), "{this} gets six charge counters"));
         Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new AddCountersTargetEffect(CounterType.CHARGE.createInstance()), new GenericManaCost(1));
         ability.addCost(new TapSourceCost());
         ability.addCost(new RemoveCountersSourceCost(CounterType.CHARGE.createInstance()));
         ability.addTarget(new TargetArtifactPermanent());
         this.addAbility(ability);
     }
 
-    public SurgeNode (final SurgeNode card) {
+    public SurgeNode(final SurgeNode card) {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/s/SurgeNode.java:37]: should be private?

**Review by xenohedron — APPROVED** [review]

File: Mage.Sets/src/mage/cards/s/SurgeNode.java (hunk 2 above)
Comment (Susucre) [reply Mage.Sets/src/mage/cards/s/SurgeNode.java:37]: Oh the space did mess with my regex. Should be at least protected indeed.

Comment (Susucre) [reply Mage.Sets/src/mage/cards/s/SurgeNode.java:37]: I'll add a reflection test for those constructor that should be non-public at some point.

**Pull request merged**